src/pal/src/include/pal/context.h

```diff
@@ -277,6 +277,57 @@ inline void *FPREG_Xstate_Ymmh(const ucontext_t *uc)
 #define MCREG_Sp(mc)      ((mc).sp)
 #define MCREG_Pc(mc)      ((mc).pc)
 #define MCREG_Cpsr(mc)    ((mc).pstate)
+
+
+#define _ALIGN_UP_16(s) ((s + 15) & ~0xf)
+inline
+fpsimd_context* GetNativeSigSimdContext(native_context_t *mc, bool createIfNotPresent)
+{
+    fpsimd_context* fp = nullptr;
+
+    size_t size = 0;
+    _aarch64_ctx* record = reinterpret_cast<_aarch64_ctx *>(&mc->uc_mcontext.__reserved[size]);
+
+    while (record->size !=0)
+    {
+        if (record->magic == FPSIMD_MAGIC)
+        {
+            fp = reinterpret_cast<fpsimd_context*>(record);
+            break;
+        }
+        size += _ALIGN_UP_16(record->size);
```

Comment:
The kernel doesn't do aligning when parsing the sigcontext reserved area:
https://elixir.bootlin.com/linux/v5.1.15/source/arch/arm64/kernel/signal.c#L488
It only asserts when the offset of the context record is not aligned.